packages/framework/src/services/microservice/gateway/kafka-gateway.service.ts

```diff
@@ -0,0 +1,30 @@
+import { CompleteTransportQueueItem, MicroserviceOptions } from '../../../models';
+import { KafkaBrokerClient } from '../../kafka/kafka-broker.client';
+import { MicroserviceGatewayHandler } from './microservice-gateway-handler.abstract';
+
+export class KafkaGatewayService extends MicroserviceGatewayHandler {
+  async publish(client: KafkaBrokerClient, item: CompleteTransportQueueItem): Promise<void> {
+    try {
+      await client.publish({ topic: item.event, messages: [JSON.stringify(item.data)] });
+      this.successPublish(item);
+    } catch (error) {
+      this.failedPublish(item);
+    }
```

Comment:
Maybe worth passing `error` to `failedPublish` so that we show as much information as possible about the failure.

Reply:
Maybe we should do a special event for errors here, but the point of `failedPublish` is to enqueue the item for automatic resend. And, for your context, the `successPublish` is to remove the item from the send queue.

Reply:
Gotcha. I guess we can tackle proper error handling as a separate pull request later on.